[Cocoa] Pass host process audit token to the WebFilterEvaluator

brentfulgham opened the pull request:
#### 6359c6e87a404d8bcd4d45bddc8aa589b4364d2e
<pre>
[Cocoa] Pass host process audit token to the WebFilterEvaluator
<a href="https://bugs.webkit.org/show_bug.cgi?id=277643">https://bugs.webkit.org/show_bug.cgi?id=277643</a>
&lt;<a href="https://rdar.apple.com/109173794">rdar://109173794</a>&gt;

Reviewed by Andy Estes.

Pass the WebKit host process audit token to the WebFilterEvaluator when parental controls
and other content filtering is enabled. This allows that framework to confirm process
entitlements and to support additional filtering features.

This patch simply passes the existing audit token value to the WebFilterEvaluator object,
when this property exists.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cocoa/WebFilterEvaluatorSPI.h:
* Source/WebCore/loader/ContentFilter.cpp:
(WebCore::ContentFilter::setHostProcessAuditToken):
* Source/WebCore/loader/ContentFilter.h:
* Source/WebCore/platform/PlatformContentFilter.h:
(WebCore::PlatformContentFilter::hostProcessAuditToken const):
(WebCore::PlatformContentFilter::setHostProcessAuditToken):
* Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
(WebCore::ParentalControlsContentFilter::responseReceived):
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::startContentFiltering):

Canonical link: <a href="https://commits.webkit.org/281923@main">https://commits.webkit.org/281923@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/85febfecfe97ad884ca28a1b67403373003ee1ea

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61408 "check-webkit-style (failure)") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40767 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13991 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65360 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11955 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63537 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48445 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12230 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49601 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8302 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64437 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37896 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53190 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30444 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34557 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10418 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10868 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/54510 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56365 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10718 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67090 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/60655 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5353 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10493 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56979 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5376 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53155 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57194 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13692 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4421 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/82423 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36571 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14387 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37654 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38748 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37398 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/loader/ContentFilter.cpp:
```diff
@@ -363,6 +363,17 @@ void ContentFilter::deliverStoredResourceData()
+void ContentFilter::setHostProcessAuditToken(std::optional<audit_token_t>&& token)
+{
+    if (!token)
+        return;
```
If we had previously set a token on `m_contentFilters` and now have `nullopt`, won't the filters be left with a stale audit token?

brentfulgham replied:
Ah, good point! No need to nullptr check, I think.

brentfulgham committed:
[Cocoa] Pass host process audit token to the WebFilterEvaluator
https://bugs.webkit.org/show_bug.cgi?id=277643
<rdar://109173794>

Reviewed by Andy Estes.

Pass the WebKit host process audit token to the WebFilterEvaluator when parental controls
and other content filtering is enabled. This allows that framework to confirm process
entitlements and to support additional filtering features.

This patch simply passes the existing audit token value to the WebFilterEvaluator object,
when this property exists.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cocoa/WebFilterEvaluatorSPI.h:
* Source/WebCore/loader/ContentFilter.cpp:
(WebCore::ContentFilter::setHostProcessAuditToken):
* Source/WebCore/loader/ContentFilter.h:
* Source/WebCore/platform/PlatformContentFilter.h:
(WebCore::PlatformContentFilter::hostProcessAuditToken const):
(WebCore::PlatformContentFilter::setHostProcessAuditToken):
* Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm:
(WebCore::ParentalControlsContentFilter::responseReceived):
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::startContentFiltering):

Canonical link: https://commits.webkit.org/281923@main
